test/e2e/vmagent_test.go

```diff
@@ -0,0 +1,281 @@
+	var vm vmInfo
+	vm.Nodename = nodeName
+	var cmd string
+	cmd = "ip -o -4 route show to default | awk '{print $5}'"
+	vm.OS = "Linux"
+	_, vm.IfName, _, err = data.provider.RunCommandOnVM(nodeName, cmd)
+	require.Nil(t, err, "Failed to run command %s on VM %s, err %v", cmd, nodeName, err)
+
+	vm.IfName = strings.TrimSpace(vm.IfName)
+	cmd = fmt.Sprintf("ifconfig %s | awk '/inet / {print $2}'| sed 's/addr://'", vm.IfName)
+	_, vm.IP, _, err = data.provider.RunCommandOnVM(nodeName, cmd)
+	require.Nil(t, err, "Failed to run command %s on VM %s, err %v", cmd, nodeName, err)
+
+	vm.IP = strings.TrimSpace(vm.IP)
+	return vm
+}
+
+func getWindowsVmInfo(t *testing.T, data *TestData, nodeName string) (vm vmInfo) {
+	var err error
+	vm.Nodename = nodeName
+	vm.OS = "Windows"
+	cmd := fmt.Sprintf("powershell 'Get-WmiObject -Class Win32_IP4RouteTable | Where { $_.destination -eq \"0.0.0.0\" -and $_.mask -eq \"0.0.0.0\"} | Sort-Object metric1 | select interfaceindex | ft -HideTableHeaders'")
+	_, ifIndex, _, err := data.provider.RunCommandOnVM(nodeName, cmd)
+	require.Nil(t, err, "Failed to run command %s on VM %s, err %v", cmd, nodeName, err)
+
+	vm.IfIndex = strings.TrimSpace(ifIndex)
+	cmd = fmt.Sprintf("powershell 'Get-NetAdapter -IfIndex %s | select name | ft -HideTableHeaders'", vm.IfIndex)
+	_, ifName, _, err := data.provider.RunCommandOnVM(nodeName, cmd)
+	require.Nil(t, err, "Failed to run command %s on VM %s, err %v", cmd, nodeName, err)
+
+	vm.IfName = strings.TrimSpace(ifName)
+	cmd = fmt.Sprintf("powershell 'Get-NetIPAddress -AddressFamily IPv4 -ifIndex %s| select IPAddress| ft -HideTableHeaders'", vm.IfIndex)
+	_, ifIP, _, err := data.provider.RunCommandOnVM(nodeName, cmd)
+	require.Nil(t, err, "Failed to run command %s on VM %s, err %v", cmd, nodeName, err)
+
+	vm.IP = strings.TrimSpace(ifIP)
+	return vm
+
+}
+
+func startAntreaAgent(t *testing.T, data *TestData, vm vmInfo) {
+	t.Logf("Starting antrea-agent on VM: %s", vm.Nodename)
+	var cmd string
+	if vm.OS == "Windows" {
+		cmd = "nssm start antrea-agent"
+	} else {
+		cmd = "sudo systemctl start antrea-agent"
+	}
+	_, _, _, err := data.provider.RunCommandOnVM(vm.Nodename, cmd)
+	require.Nil(t, err, "Failed to run command %s on VM %s, err %v", cmd, vm.Nodename, err)
+}
+
+func stopAntreaAgent(t *testing.T, data *TestData, vm vmInfo) {
+	t.Logf("Stopping antrea-agent on VM: %s", vm.Nodename)
+	var cmd string
+	if vm.OS == "Windows" {
+		cmd = "nssm stop antrea-agent"
+	} else {
+		cmd = "sudo systemctl stop antrea-agent"
+	}
+	_, _, _, err := data.provider.RunCommandOnVM(vm.Nodename, cmd)
+	require.Nil(t, err, "Failed to run command %s on VM %s, err %v", cmd, vm.Nodename, err)
+}
+
+func verifyInterfaceIsInOVS(t *testing.T, data *TestData, vm vmInfo) (found bool, err error) {
+	var cmd string
+	if vm.OS == "Windows" {
+		cmd = fmt.Sprintf("ovs-vsctl --column=name list port '%s'", vm.IfName)
+	} else {
+		cmd = fmt.Sprintf("sudo ovs-vsctl --column=name list port %s", vm.IfName)
```

Comment:
Is it working on Linux if we use `'%s'` like the command on Windows?

Reply:
Yes. the test is passing for both windows and linux